File: javascript/armstrong-numbers/armstrong-numbers.js
export const isArmstrongNumber = (number) => {
  const digits = [...number + ''].map((digit) => parseInt(digit));
  const powers = digits.map((digit) => digit ** digits.length);
  const armstrong = powers.reduce((value, sum) => value + sum);
  return number === armstrong;
};

File: java/armstrong-numbers/src/main/java/ArmstrongNumbers.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

class ArmstrongNumbers {
    boolean isArmstrongNumber(int numberToCheck) {
        String numberString = String.valueOf(numberToCheck);
        int length = numberString.length();

        String[] chars = numberString.split("");
        List<String> digits = new ArrayList<>(Arrays.asList(chars));

        Double armstrong = digits.stream()
                .map((digit) -> Math.pow(Integer.parseInt(digit), length))
                .reduce((double) 0, Double::sum);

        return armstrong == numberToCheck;
    }
}
